manager/scheduler/scheduler.go

```diff
@@ -242,13 +242,17 @@ func (s *Scheduler) deleteTask(ctx context.Context, t *api.Task) {
 }
 
 func (s *Scheduler) createOrUpdateNode(n *api.Node) {
-	var resources api.Resources
+	nodeInfo := s.nodeHeap.nodeInfo(n.ID)
 	if n.Description != nil && n.Description.Resources != nil {
-		resources = *n.Description.Resources
+		if nodeInfo.AvailableResources == nil {
+			// if nodeInfo.AvailableResources hasn't been initialized
+			// we copy resources information from node description and
+			// pass it to nodeInfo
+			resources := *n.Description.Resources
+			nodeInfo.AvailableResources = &resources
```

Comment:
This could be `nodeInfo.AvailableResources = n.Description.Resources.Copy()`, but the above is fine too.